subsys/bluetooth/controller/ll_sw/ctrl.c

```diff
@@ -709,8 +712,68 @@ static u32_t isr_rx_adv_sr_report(struct pdu_adv *pdu_adv_rx, u8_t rssi_ready)
 }
 #endif /* CONFIG_BLUETOOTH_CONTROLLER_SCAN_REQ_NOTIFY */
 
-static inline u32_t isr_rx_adv(u8_t devmatch_ok, u8_t irkmatch_ok,
-			       u8_t irkmatch_id, u8_t rssi_ready)
+static inline bool isr_adv_sr_check(struct pdu_adv *pdu, u8_t devmatch_ok,
+				    u8_t irkmatch_ok, u8_t irkmatch_id)
+{
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_PRIVACY)
+	return ((((_radio.advertiser.filter_policy & 0x01) == 0) &&
+		 ctrl_rl_allowed(pdu->tx_addr,
```

Comment:
Use (!ctrl_rl_enabled() || ctrl_rl_allowed(...))... see my other comment in ctrl_rl_allowed definition.